### ISSUE TITLE
Very incomplete diacritics support

### ISSUE THREAD

**Issue by TempaYeshe** [tadashi-aikawa/obsidian-various-complements-plugin#310]
Hi,
Right now, you are using [this solution](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/blob/main/src/util/diacritics-map.ts) to remove diacritics for the "Treat accent diacritics as alphabetic characters" option. 
The problem is that it does not cover all diacritic combinations at all, and it does not look very clean. There is a much better solution, used in [this other plugin](https://github.com/tth05/obsidian-completr/commit/a185f47cfae79ec074cc102528ece06a7d9e87c0), which is:
1. "Decomposing" the diacritics with the [String.prototype.normalize("NFD")](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/String/normalize) function. The result in unicode is the base letter without diacritic code + the appropriate ["Combining Diacritical Mark"](https://en.wikipedia.org/wiki/Combining_Diacritical_Marks) code(s). 
2. Now, you just have to remove the combining diacritical mark(s) with a regular expression `/[\u0300-\u036f]/g`

So in the end there is a very simple function to remove all possible diacritics ([taken from here](https://github.com/tth05/obsidian-completr/commit/a185f47cfae79ec074cc102528ece06a7d9e87c0)):

```
function removeDiacritics(str: string): string {
    return str.normalize("NFD").replace(/[\u0300-\u036f]/g, "");
}
```

Thank you for your plugin!

**Comment by tadashi-aikawa** [comment]
Thank you for your suggestion, @TempaYeshe :)

This is about a different plugin, but I once implemented it in exactly the same way as you suggested.

https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/commit/cea95e6ad602e73f8cede9462da4b6c034a6ef84

However, I received reports that this implementation does not handle certain symbols. After that, I rewrote it to cover those as well, which led to the current somewhat forceful implementation.

https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/issues/23

Unfortunately, since I never use diacritics, I am unsure about what is reasonable or what to prioritize for this specification. If you happen to know a solution that could resolve the issue above, I would appreciate it if you could share it with me👍

**Comment by TempaYeshe** [comment]
The technical reason is that "Ø" is not a diacritic (cf. [Wikipedia](https://en.wikipedia.org/wiki/Ø#Typing_the_character)), like other characters like "œ". Right now your current solution has the benefit of covering these cases that are not "pure" diacritics, so I guess it is not worth changing. But there is a problem with it when a word starts with one or multiple diacritics such as `śūnyatā`: the completion does not trigger if you only type the letters without diacritics (`s` in the example, you need to type explicitly `śū` which defeats the purpose).

**Comment by tadashi-aikawa** [comment]
@TempaYeshe 
Thank you for the detailed explanation. I wasn't aware of the distinction between pure diacritics and those that aren't, so I learned something new.

> But there is a problem with it when a word starts with one or multiple diacritics such as śūnyatā: the completion does not trigger if you type the letters without diacritics

I tried writing a test using śūnyatā for the exact same method in a different project, and contrary to my expectations, the test passed successfully. Since Various Complements use the same logic, I believe sunyata should also work.

![image](https://github.com/user-attachments/assets/597b4abd-7922-449c-84fa-4ee76ec43273)

![image](https://github.com/user-attachments/assets/b9801558-536d-4c38-846e-6a615bac5998)

If you encounter any issues in your environment, it would be helpful if you could share the reproduction steps within Sandbox Vault.

**Comment by TempaYeshe** [comment]
Thank you for helping me. The problem happens in the sandbox vault (Obsidian v1.6.7) with only Various Complements and completely default settings:

https://github.com/user-attachments/assets/1fb8c004-43fd-4415-9835-bf75a32702f7

**Comment by tadashi-aikawa** [comment]
Thank you, @TempaYeshe :)

I could reproduce it 👍

**Comment by tadashi-aikawa** [comment]
@TempaYeshe 
I have identified the cause. This issue occurs only when the "Current file complement" is used, and the "Match strategy" is set to "prefix."

I have a fix in sight and plan to release a revised version by the end of this week.

**Comment by tadashi-aikawa** [comment]
@TempaYeshe 
Released in [v10.0.3](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/10.0.3) 🚀

**Comment by TempaYeshe** [comment]
Works wonderfully, thank you!
